src/client/java/teammates/client/scripts/sql/DataMigrationForTeamSql.java

```diff
@@ -0,0 +1,87 @@
+package teammates.client.scripts.sql;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.googlecode.objectify.cmd.Query;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.JoinType;
+import jakarta.persistence.criteria.Root;
+
+import teammates.common.util.HibernateUtil;
+import teammates.storage.entity.Course;
+import teammates.storage.entity.CourseStudent;
+import teammates.storage.sqlentity.Section;
+import teammates.storage.sqlentity.Team;
+
+/**
+ * Data migration class for team entity.
+ */
+@SuppressWarnings("PMD")
+public class DataMigrationForTeamSql extends
+        DataMigrationEntitiesBaseScriptSql<Course, teammates.storage.sqlentity.Team> {
+
+    public static void main(String[] args) {
+        new DataMigrationForTeamSql().doOperationRemotely();
+    }
+
+    @Override
+    protected Query<Course> getFilterQuery() {
+        return ofy().load().type(Course.class);
+    }
+
+    @Override
+    protected boolean isPreview() {
+        return false;
+    }
+
+    /*
+     * Sets the migration criteria used in isMigrationNeeded.
+     */
+    @Override
+    protected void setMigrationCriteria() {
+        // No migration criteria currently needed.
+    }
+
+    @Override
+    protected boolean isMigrationNeeded(Course entity) {
+        return true;
+    }
+
+    @Override
+    protected void migrateEntity(Course oldCourse) throws Exception {
+        HibernateUtil.beginTransaction();
+        teammates.storage.sqlentity.Course newCourse = getNewCourse(oldCourse.getUniqueId());
+        Map<String, Section> sectionNameToSectionMap =
+                newCourse.getSections().stream().collect(Collectors.toMap(Section::getName, s -> s));
+
+        getTeamNameToSectionNameMap(oldCourse).forEach((teamName, sectionName) ->
+                        HibernateUtil.persist(new Team(sectionNameToSectionMap.get(sectionName), teamName)));
+
+        HibernateUtil.commitTransaction();
+    }
+
+    private teammates.storage.sqlentity.Course getNewCourse(String courseId) {
```

Comment:
Great job on this and the benchmarking, wondering if we can abstract this out into another file as a helper method as `getNewCourseWithTeams` and make use of the optimization in `VerifyTeamAttributes` too?

Would a hypothetical further optimization be a 3rd join on teams in `VerifyTeamAttributes`? Not needed now, but would be great if can note it down as a comment too

Reply:
Or would calling this method in `VerifyTeamAttributes` be redundant, since the teams would already have been fetched when verifying sections?